infrastructure/server/admin_management_api_gateway.tf

```diff
@@ -0,0 +1,103 @@
+resource "aws_api_gateway_rest_api" "admin_management_api_gateway_rest_api" {
+  name = "${aws_lambda_function.fam_admin_management_api_function.function_name}-gateway"
+  description = "Proxy API Gateway to handle request to fam-admin-management-api-lambda."
+  endpoint_configuration {
+    types = ["REGIONAL"]
+  }
+
+  tags = {
+    Name = "fam_admin_api_gateway"
+    managed-by = "terraform"
+  }
+}
+
+resource "aws_api_gateway_resource" "admin_management_api_gateway_resource_proxy" {
+  rest_api_id = aws_api_gateway_rest_api.admin_management_api_gateway_rest_api.id
+  parent_id   = aws_api_gateway_rest_api.admin_management_api_gateway_rest_api.root_resource_id
+  path_part   = "{proxy+}"
+}
+
+resource "aws_api_gateway_method" "admin_management_api_gateway_method_proxy" {
+  rest_api_id   = aws_api_gateway_rest_api.admin_management_api_gateway_rest_api.id
+  resource_id   = aws_api_gateway_resource.admin_management_api_gateway_resource_proxy.id
+  http_method   = "ANY"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "admin_management_api_gateway_integration_proxy" {
+  rest_api_id = aws_api_gateway_rest_api.admin_management_api_gateway_rest_api.id
+  resource_id = aws_api_gateway_resource.admin_management_api_gateway_resource_proxy.id
+  http_method = aws_api_gateway_method.admin_management_api_gateway_method_proxy.http_method
+
+  # "...specifying how API Gateway will interact with the back end."
+  # Required if type is AWS_PROXY (and some other types).
+  # Lambda function can only be invoked via POST.
+  integration_http_method = "POST"
+  type                    = "AWS_PROXY"
+  uri                     = aws_lambda_function.fam_admin_management_api_function.invoke_arn
+
+  depends_on = [
+    aws_api_gateway_method.admin_management_api_gateway_method_proxy,
+    aws_lambda_function.fam_admin_management_api_function
+  ]
+}
+
+resource "aws_api_gateway_deployment" "admin_management_api_gateway_deployment" {
+  rest_api_id = aws_api_gateway_rest_api.admin_management_api_gateway_rest_api.id
+
+  depends_on = [
+    aws_api_gateway_integration.admin_management_api_gateway_integration_proxy
+  ]
+
+  triggers = {
+    redeployment = sha1(jsonencode([
+      aws_api_gateway_resource.admin_management_api_gateway_resource_proxy.id,
+      aws_api_gateway_method.admin_management_api_gateway_method_proxy.id,
+      aws_api_gateway_integration.admin_management_api_gateway_integration_proxy.id
+    ]))
+    # https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment mentioned
+    # using whole resouce with filesha1() that may be consider in the future.
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_api_gateway_stage" "admin_management_api_gateway_stage" {
+  deployment_id = aws_api_gateway_deployment.admin_management_api_gateway_deployment.id
+  rest_api_id   = aws_api_gateway_rest_api.admin_management_api_gateway_rest_api.id
+  stage_name    = "v1"
```

Comment:
I always found this 'v1' stage name weird. I'm not sure what the AWS docs recommend. I have a vague recollection of talking to Conrad about this but I don't remember the details.

Reply:
Thanks Basil! I was also not certain where to see the usefulness of this "stage_name" (**v1**) field come in but didn't ask this question. Simple I just reuse what it was from original first api-gateway.
I guess this has something to do with **pipeline** (dev/test/prod?) deployment but we don't really need this, since we have different AWS space for deployments; however, it is a required field.
The other guess is if we have some lambda api changes (second version **v2**) that needed to be tested on before making it official, it's easy to create different "stage_name" (v2?) for experimental and only allow for restrict testing before switching from v1 to v2 I guess.
Anyways, I just coded it the same as first api-gateway. If you feel we can make this name more meaningful I will change that name to your suggestion.
By the way, this "stage" is now as a **Terraform resource** now, rather than "**stage_name**" in api_gateway_deployment's attribute (as Terraform suggested), so that's different than the first api-gateway; and it works.

Reply:
I looked into the AWS docs. To quote "a stage is a named reference to a deployment, which is a snapshot of the API. You use a Stage to manage and optimize a particular deployment". 

I wasn't able to come up with any better name, so might as well leave it v1, at least that's consistent with the other gateway.